### PR TITLE
Improvements to Datasources and GridMet Content

### PR DESCRIPTION
**Pull request by meghanakr7**
1. Added images for dataset
2. Executed all the cells
3. Added datset Download code for gridMET and SNOTEL
4. Added DEM datasource
5. Resolved comments

### CODE REVIEW

File: book/chapters/datasource.ipynb [hunk 1]
```diff
@@ -139,25 +395,58 @@
...
+    "        current_date += delta\n",
+    "\n",
+    "    return links\n",
+    "\n",
+    "if __name__ == \"__main__\":\n",
+    "    start_year = 2019\n",
+    "    end_year = 2022\n",
+    "\n",
+    "    links = generate_links(start_year, end_year)\n",
+    "    save_location = \"/Users/meghana/gridmet_test_run/amsr\"\n",
```
Comment (gokulprathin8) [inline book/chapters/datasource.ipynb]: can you please remove absolute path? and make it relative
also commit the data that is being used to the repository.

File: book/chapters/datasource.ipynb [hunk 2]
```diff
@@ -313,6 +585,424 @@
...
+      "</html>\n",
+      "\n",
+      "Container div not found\n",
+      "None\n"
+     ]
+    }
+   ],
+   "source": [
+    "from datetime import datetime\n",
+    "from metloom.pointdata import SnotelPointData\n",
```
Comment (gokulprathin8) [inline book/chapters/datasource.ipynb]: I dont think we are using this. please remove commented code.

File: book/chapters/datasource.ipynb [hunk 3]
```diff
@@ -313,6 +585,424 @@
     "\n",
     "For more information, visit the [NRCS SNOTEL page](https://www.nrcs.usda.gov/wps/portal/wcc/home/aboutUs/monitoringPrograms/automatedSnowMonitoring).\n"
```
Comment (gokulprathin8) [inline book/chapters/datasource.ipynb:605]: we can add a image from usda here
which you can find here
https://nwcc-apps.sc.egov.usda.gov/imap/#version=169&elements=&networks=!&states=!&basins=!&hucs=&minElevation=&maxElevation=&elementSelectType=any&activeOnly=true&activeForecastPointsOnly=false&hucLabels=false&hucIdLabels=false&hucParameterLabels=true&stationLabels=&overlays=&hucOverlays=2&basinOpacity=75&basinNoDataOpacity=25&basemapOpacity=100&maskOpacity=0&mode=data&openSections=dataElement,parameter,date,basin,options,elements,location,networks&controlsOpen=true&popup=&popupMulti=&popupBasin=&base=esriNgwm&displayType=basinstation&basinType=6&dataElement=WTEQ&depth=-2&parameter=PCTMED&frequency=DAILY&duration=I&customDuration=&dayPart=E&monthPart=E&forecastPubDay=1&forecastExceedance=50&useMixedPast=true&seqColor=1&divColor=7&scaleType=D&scaleMin=&scaleMax=&referencePeriodType=POR&referenceBegin=1991&referenceEnd=2020&minimumYears=20&hucAssociations=true&relativeDate=-1&lat=41.691&lon=-112.200&zoom=6.0

File: book/chapters/datasource.ipynb [hunk 4]
```diff
@@ -22,6 +22,14 @@
     "* The dataset provides an extensive array of information, including maximum and minimum temperatures, precipitation, radiation, wind velocity, and various other meteorological variables.\n",
     "* Scientists and modelers find it invaluable for a myriad of purposes, such as ecological research, agricultural strategizing, hydrological modeling, and beyond.\n",
     "\n",
+    "<p align=\"center\">\n",
+    "  <img src=\"https://developers.google.com/earth-engine/datasets/images/IDAHO_EPSCOR/IDAHO_EPSCOR_GRIDMET_sample.png\" alt=\"Sample Image from IDAHO_EPSCOR_GRIDMET dataset\" width=\"400\">\n",
+    "</p>\n",
+    "<p align=\"center\"> IDAHO EPSCOR GRIDMET dataset</p>\n",
```
Comment (gokulprathin8) [inline book/chapters/datasource.ipynb]: adding images to each dataset will make it look more attractive
you can copy paste images from here

https://docs.google.com/presentation/d/1-dGIHeza5dOCRW3UVR-lTGa1b7QPUiXi8Jn_9WjO05k/edit?usp=sharing

File: book/chapters/datasource.ipynb [hunk 5]
```diff
@@ -313,6 +585,424 @@
...
+    "print(container_div)\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "8ee2035e",
+   "metadata": {},
+   "source": [
+    "We automate the process of accessing and parsing NOAA's snow data using BeautifulSoup. By constructing a query URL based on user-defined location and date parameters, we automate the process of fetching web content and parsing HTML to extract pertinent information. The above script enhances efficiency in obtaining snow data, offering a user-friendly approach for various analyses or applications without the need for technical expertise."
+   ]
```
Comment (gokulprathin8) [inline book/chapters/datasource.ipynb:1021]: I think DEM is missing
Can you please add it.

File: book/chapters/gridmet.ipynb [hunk 6]
```diff
@@ -22,7 +22,7 @@
   },
```
Comment (gokulprathin8) [inline book/chapters/gridmet.ipynb:22]: the ipython json is too long here in github to review correctly. I'll just type it out here instead.

image
I dont think we need map visualization logic here in gridmet section.
This needs to be at the last but 1 chapter before conclusion.

File: book/chapters/gridmet.ipynb [hunk 7]
```diff
@@ -22,7 +22,7 @@
...
   {
```
Comment (gokulprathin8) [inline book/chapters/gridmet.ipynb:23]: image

can you change this to "Changing the timeframe from 10 years to 1 year to demonstrate"
currently it is reverse.

File: book/chapters/gridmet.ipynb [hunk 8]
```diff
@@ -22,7 +22,7 @@
...
    "cell_type": "code",
```
Comment (gokulprathin8) [inline book/chapters/gridmet.ipynb]: image


please remove commented code.

**Review by gokulprathin8 — CHANGES_REQUESTED** [review]
minor changes.

File: book/chapters/gridmet.ipynb (hunk 8 above)
Comment (meghanakr7) [reply book/chapters/gridmet.ipynb]: done

File: book/chapters/gridmet.ipynb (hunk 6 above)
Comment (meghanakr7) [reply book/chapters/gridmet.ipynb:22]: done

File: book/chapters/datasource.ipynb (hunk 1 above)
Comment (meghanakr7) [reply book/chapters/datasource.ipynb]: done

File: book/chapters/gridmet.ipynb (hunk 7 above)
Comment (meghanakr7) [reply book/chapters/gridmet.ipynb:23]: done

File: book/chapters/datasource.ipynb (hunk 2 above)
Comment (meghanakr7) [reply book/chapters/datasource.ipynb]: done

File: book/chapters/datasource.ipynb (hunk 5 above)
Comment (meghanakr7) [reply book/chapters/datasource.ipynb:1021]: done

File: book/chapters/datasource.ipynb (hunk 3 above)
Comment (meghanakr7) [reply book/chapters/datasource.ipynb:605]: done

File: book/chapters/datasource.ipynb (hunk 4 above)
Comment (meghanakr7) [reply book/chapters/datasource.ipynb]: done

**Review by gokulprathin8 — APPROVED** [review]
LTGM
Good Job!

**Pull request merged**